docs/adr/adr-001-predevnet-celestia-node.md

```diff
@@ -0,0 +1,131 @@
+    * `ExtendedHeaderVerification` (`light` nodes only)
+    * `ExtendedHeaderSub`
+    * `ExtendedHeaderStore`
+* `FraudProofService` *(optional for devnet)*
+    * **`FraudProofGeneration`**
+    * `FraudProofSub`
+    * `FraudProofStore`
+* `ShareService`
+    * `ShareExchange`
+    * `ShareStore`
+* `BlockService`
+    * `BlockErasureCoding` 
+    * `NewBlockEventSubscription` (`full` node <> `Celestia Core` node)
+    * `BlockExchange` (`full` node <> `full` node)
+    * `BlockStore`
+* `StateService` *(optional for devnet)*
+    * `StateExchange`
+* `TransactionService` *(dependent on `StateService` implementation, but optional for devnet)*
+    * `SubmitTx`\
+
+
+For devnet, it should be possible for Celestia `full` Nodes to receive information directly from Celestia Core nodes or from each other. 
+
+## Considerations 
+
+### State Fraud Proofs
+For the Celestia Node to be able to propagate `StateFraudProof`s, we must modify Celestia Core to store blocks with invalid state and serve them to both the Celestia Node and the Celestia App, **and** the Celestia App must be able to generate and serve `StateFraudProof`s via RPC to Celestia nodes.
+
+This feature is not necessarily required for devnet (so state exection functionality for Celestia Full Nodes can be stubbed out), but it would be nice to have for devnet as we will likely allow Celestia Full Nodes to speak with other Celestia Full Nodes instead of running a trusted Celestia Core node simultaenously and relying on it for information.
+
+A roadmap to implementation could look like the following: 
+
+The Celestia Full Node would adhere to the ABCI interface in order to communicate with the Celestia App (similar to the way Optimint does it). The Celestia Full Node would send State requests to the Celestia App in order for the Celestia app to replay the transactions in the block and verify the state. 
+
+For devnet, it is okay to stub out state verification functionality. For example, a Celestia Full Node would download reserve transactions, but not replay them. 
+
+
+### Fraud Proofs
+
+For validators (Celestia Core) to be able to sign blocks, they need to erasure code the block data and commit to the data root of the erasure coded block that is stored in the header. The validators do not need to store the erasure coded block, but only the "raw" block with the data root in the header. A Celestia Node will fetch the "raw" block from Celestia Core via the `/block` endpoint, and then erasure code the block, checking it against the data root in the header, and then store the erasure coded block.
+
+For devnet, the Celestia Node will not be able to generate state fraud proofs as it will not have a state execution environment, so it will rely on Celestia Core to produce state fraud proofs and retrieve the proof via a `/state_fraud` endpoint. The Celestia Node will also require the "state fraudulent" block, so Celestia Core needs to make those blocks available via the `/block` endpoint.
```

Comment:
Let's put this on the backburner but it might make sense that the Celestia node will have to generate the state fraud proof itself. That way, we might not need to modify tendermint but instead, the Celestia node will get the header and the reserved tx from other Celestia nodes (via libp2p instead of from Tendermint via rpc) and generate the state fraud-proof itself.

Reply:
Okay but then this requires us implementing the ABCI interface and communicating with the Celestia App -- should this be a goal before devnet?

Reply:
Again, not necessarily. Of course the more we get done for devnet, the better but it's not a must-have.